backend/tests/test_callbacks.py

```diff
@@ -107,6 +107,65 @@ async def auth_func(
         assert result is None
 
 
+async def test_custom_authenticate_user(test_config):
+    from unittest.mock import patch
+
+    from chainlit.callbacks import custom_authenticate_user
+    from chainlit.user import User
+
+    # Mock the get_configured_oauth_providers function
+    with patch(
+        "chainlit.callbacks.get_configured_oauth_providers",
+        return_value=["custom_provider"],
+    ):
+
+        @custom_authenticate_user
+        async def auth_func(
+            provider_id: str,
+            token: str,
+            raw_user_data: dict,
+            default_app_user: User,
+            id_token: str | None = None,
+        ) -> User | None:
```

Comment:
What's the reason to add this callback over using `@oauth_callback`?

What does the value of `id_token` represent, in OAuth terms -- and why is it not tested?